src/components/EditCoursePage/CollapsibleCourseRun.jsx

```diff
@@ -87,10 +87,54 @@ class CollapsibleCourseRun extends React.Component {
 
     this.state = {
       open: false,
+      returnFromStaff: false,
     };
 
     this.openCollapsible = this.openCollapsible.bind(this);
     this.setCollapsible = this.setCollapsible.bind(this);
+    this.setFocus = this.setFocus.bind(this);
+  }
+
+  componentDidMount() {
+    const {
+      stafferInfo: { returnToEditCourse },
+      sourceInfo: { referringRun },
+      courseRun: { key },
+    } = this.props;
+
+    if (returnToEditCourse) {
+      if (referringRun === key) {
+        this.openCollapsible();
+        this.setFocus(returnToEditCourse);
+      }
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const {
+      open,
+      returnFromStaff,
+    } = this.state;
+    const {
+      courseId,
+    } = this.props;
+    const {
+      returnFromStaff: wasReturnedFromStaff,
+    } = prevState;
+
+    if (open && returnFromStaff) {
+      this.setFocus(false);
+    }
+    if (wasReturnedFromStaff && !returnFromStaff) {
+      const element = document.getElementById(`${courseId}.staff.label`);
+      element.scrollIntoView({ behavior: 'smooth', block: 'start', inline: 'nearest' });
+    }
+  }
+
+  setFocus(focus) {
+    this.setState({
+      returnFromStaff: focus,
+    });
```

Comment:
This method name is very genericly named, but the state it toggles is very specificly named. Maybe generic-ize the state name or specific-ize the method name?

Reply:
+1